de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/JsonHandler.java

```diff
@@ -67,22 +77,53 @@ public void applyJoinRelation(FskPortObject fskObj, List<JoinRelationAdvanced> j
             String resourcePath = sourceModel.getGeneratedResourcesDirectory().get()
                 .getAbsolutePath().replaceAll("\\\\", "/") + "/";
             String jsonPath = resourcePath + JSON_FILE_NAME;
-            loadParametersIntoWorkspace(jsonPath, sourceParameter, param.getId());
-
-            // if target parameter is of type FILE, add path to generatedResources to sourceParam
-            // This should be safe since source and target parameter must have the same type (if
-            // File)
-            if (param.getDataType().equals(Parameter.DataTypeEnum.FILE)) {
-              addPathToFileParameter(param.getId(), resourcePath);
+            if(!sourceTargetPathMap.containsKey(jsonPath)) {

```

Comment:
This is more a matter of style. If sourceTargetPath does not contain jsonPath a new List of JoinerObject is added. In any case a new JoinerObject is added to the list.

Here I would create the joiner object first because it'll be used in both cases.
```java
JoinerObject joinerObject = new JoinerObject(sourceParameter,param,joinRelation, resourcePath)
```
Then I'd just add joinerObject or initialize the list with it according to the if:
```java
if (!sourceTargetPathMap.containsKey(jsonPath)) {
    sourceTargetPathMap.put(jsonPath, Arrays.asList(joinerObject));
} else {
    sourceTargetPathMap.get(jsonPath).add(joinerObject);
}
```